### PR TITLE
Migrate to Eslint, uninstall tslint and fix some issues

### PR DESCRIPTION
**Pull request by yogmel**
This fixes issue #13

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (iamtraction) [inline .vscode/settings.json]: You probably added this file by mistake. It should be removed.

File: package.json [hunk 2]
```diff
@@ -8,10 +8,10 @@
   "repository": "https://github.com/AwesomeDocs/AwesomeDocs",
   "scripts": {
     "build": "./node_modules/.bin/gatsby build",
-    "lint": "./node_modules/.bin/tslint -p tsconfig.json",
     "start": "./node_modules/.bin/gatsby develop",
     "test": "yarn lint && yarn transpile",
-    "transpile": "./node_modules/.bin/tsc"
+    "transpile": "./node_modules/.bin/tsc",
+    "lint": "yarn eslint . --ext .js --ext .ts --fix"
```
Comment (iamtraction) [inline package.json]: This won't lint any of the files, use this instead:
```suggestion
    "lint": "./node_modules/.bin/eslint . --ext .js,.jsx,.ts,.tsx"
```

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+gatsby-config.js
+gatsby-browser.js
+gatsby-node.js
```
Comment (iamtraction) [inline .eslintignore]: These files should be linted too, instead of being ignored.

File: .eslintrc.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+extends:
+  - 'eslint:recommended'
+  - 'plugin:@typescript-eslint/eslint-recommended'
+  - 'plugin:@typescript-eslint/recommended'
+globals:
+  Atomics: readonly
+  SharedArrayBuffer: readonly
+parser: '@typescript-eslint/parser'
+parserOptions:
+  ecmaVersion: 2018
```
Comment (iamtraction) [inline .eslintrc.yml]: Bump ECMAScript version to 2020:
```suggestion
  ecmaVersion: 2020
```

**Review by iamtraction — CHANGES_REQUESTED** [review]
Apart from the suggestions I've made, why was the `tsconfig.json` file removed? It isn't related to TSLint. It's for configuring the TypeScript interpreter.

**Comment by yogmel** [comment]
@k3rn31p4nic sorry I have never implemented Eslint before. Hopefully it looks better now.

**Comment by iamtraction** [comment]
No problem.

**Review by iamtraction — APPROVED** [review]

**Pull request merged**